test/built-ins/RegExp/prototype/Symbol.match/u-lastindex.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: lastIndex is advanced according to width of astral symbols
+es6id: 21.2.5.6
+info: >
+    21.2.5.6 RegExp.prototype [ @@match ] ( string )
+
+    [...]
+    8. Else global is true,
+       a. Let fullUnicode be ToBoolean(Get(rx, "unicode")).
+       [...]
+       g. Repeat,
+          [...]
+          iv. Else result is not null,
+              [...]
+              5. If matchStr is the empty String, then
+                 [...]
+                 c. Let nextIndex be AdvanceStringIndex(S, thisIndex,
+                    fullUnicode).
+                 d. Let setStatus be Set(rx, "lastIndex", nextIndex, true).
+features: [Symbol.match]
+---*/
+
+var r = /(𝌆)?/ug;
+r[Symbol.match]('𝌆');
+assert.sameValue(r.lastIndex, 2);
```

Comment:
Ah, okay. So I think this behavior can't be externally verified through inspection of the `lastIndex` property. What about this instead:

`test/built-ins/RegExp/prototype/Symbol.match/u-lastindex.js`, renamed to `test/built-ins/RegExp/prototype/Symbol.match/u-advance-after-empty.js`

```js
var match = /^|\udf06/ug[Symbol.match]('\ud834\udf06');

assert(match !== null);
assert.sameValue(match.length, 1);
assert.sameValue(match[0], '');
```

`test/built-ins/RegExp/prototype/Symbol.replace/u-lastindex.js`, renamed to `test/built-ins/RegExp/prototype/Symbol.replace/u-advance-after-empty.js`

```js
var str = /^|\udf06/ug[Symbol.replace]('\ud834\udf06', 'XXX');
assert.sameValue(str, 'XXX\ud834\udf06');
```